extensions/Redis/Redis/RedisMemory.cs

```diff
@@ -111,8 +111,12 @@ public async Task<string> UpsertAsync(string index, MemoryRecord record, Cancell
     {
         var normalizedIndexName = NormalizeIndexName(index, this._config.AppPrefix);
         var key = Key(normalizedIndexName, record.Id);
```

Comment:
should we validate that `record.Id` is valid, e.g. no trailing spaces, not empty, no invalid chars that could inject LUA code, to avoid errors/code injection later when passing `key` to the LUA script?

Reply:
The wire protocol for Redis [RESP](https://github.com/redis/redis-specifications/blob/master/protocol/RESP2.md)(Redis Serialization Protocol) automatically parameterizes the arguments passed into every command, so the `EVAL`/`EVALSHA` commands get `KEYS` and `ARGV` via parameterized lists of arguments. So let's take this example (stripping out payload/embedding for readability):

```
"EVALSHA" "ae03b246bcbe556aec893e033a92632cd6c03e6c" "1" "tests-default:d=2//p=a2a9371108ec466f86d1a0a37cd64750" "tests-default" "embedding" "your_embedding" "__file_part" "3bf34b3500a64953834c16eb2e4c1fd9" "user" "flash" "payload" "your_payload"
```

What's actually being sent over the wire to redis looks like this:

```
*13\r\n
$7\r\n
EVALSHA\r\n
$40\r\n
ae03b246bcbe556aec893e033a92632cd6c03e6c\r\n
$1\r\n
$53\r\n
tests-default:d=2//p=a2a9371108ec466f86d1a0a37cd64750\r\n
$13\r\n
tests-default\r\n
$9\r\n
embedding\r\n
$14\r\n
your_embedding\r\n
$11\r\n
__file_part\r\n
$32\r\n
3bf34b3500a64953834c16eb2e4c1fd9\r\n
$4\r\n
user\r\n
$5\r\n
flash\r\n
$7\r\n
payload\r\n
$12\r\n
your_payload\r\n
```

So in general injection attacks aren't really possible if you are controlling the interaction to the command interface. There have been CVEs where it was possible to use a LUA script to do nefarious things in Redis (e.g. crash Redis or privileged access to it), but that assumes the attacker can run their own scripts.